inst/tests/mergelist.Rraw

```diff
@@ -372,6 +377,908 @@ local({
+test(101.38, mergelist(list(data.table(id1=c(3:2, 2L), v1=1:3), data.table(id1=2:1, v2=2:3)), how="full", on="id1", mult="error"), error="multiple matches during merge") ## how="full" mult="error" confirm that second binary merge anti join 'bns', is covering both sides error, this test doesnt ensure about that but serves a data that has been used for debugging with extra 'cat("before|after bns\n")'
+test(101.39, mergelist(list(data.table(x=1L), data.table(x=1L)), on="x", mult="bad"), error="'mult' must be one of [error, all, first, last] or NULL, or a list")
+
+## how= default & invalid
+test(101.40, mergelist(list(data.table(x=1:2), data.table(x=1L, v2=1L)), on="x"), data.table(x=1:2, v2=c(1L, NA))) ## match.arg sets how="left", only when missing or NULL, otherwise we do own check
+test(101.41, mergelist(list(data.table(x=1:2), data.table(x=1L, v2=1L)), on="x", how=NULL), data.table(x=1:2, v2=c(1L, NA))) ## still match.arg
+test(101.42, mergelist(list(data.table(x=1L), data.table(x=1L)), on="x", how="bad"), error="'how' must be one of [left, inner, full, right, semi, anti, cross], or a list")
+test(101.43, mergelist(list(data.table(x=1L), data.table(x=1L)), on="x", how=list("bad")), error="'how' must be one of [left, inner, full, right, semi, anti, cross], or a list")
+
+## missing on argument
+### neither table keyed
+local({
+  l = list(data.table(x=1:2), data.table(x=2:3))
+  test(102.01, mergelist(l, how="inner"), error="necessary key is not present")
+  test(102.02, mergelist(l, how="left"), error="necessary key is not present")
+  test(102.03, mergelist(l, how="right"), error="necessary key is not present")
+  test(102.04, mergelist(l, how="full"), error="necessary key is not present")
+})
+### left table keyed
+local({
+  l = list(data.table(x=1:2, key="x"), data.table(x=2:3))
+  test(102.11, mergelist(l, how="inner"), data.table(x=2L, key="x"))
+  test(102.12, mergelist(l, how="left"), error="necessary key is not present")
+  test(102.13, mergelist(l, how="right"), data.table(x=2:3))
+  test(102.14, mergelist(l, how="full"), data.table(x=1:3))
+})
+### right table keyed
+local({
+  l = list(data.table(x=1:2), data.table(x=2:3, key="x"))
+  test(102.21, mergelist(l, how="inner"), data.table(x=2L))
+  test(102.22, mergelist(l, how="left"), data.table(x=1:2))
+  test(102.23, mergelist(l, how="right"), error="necessary key is not present")
+  test(102.24, mergelist(l, how="full"), data.table(x=1:3))
+})
+### both tables keyed
+local({
+  l = list(data.table(x=1:2, key="x"), data.table(x=2:3, key="x"))
+  test(102.31, mergelist(l, how="inner"), data.table(x=2L, key="x")) ## ordered subset
+  test(102.32, mergelist(l, how="left"), data.table(x=1:2, key="x"))
+  test(102.33, mergelist(l, how="right"), data.table(x=2:3, key="x"))
+  test(102.34, mergelist(l, how="full"), data.table(x=1:3))
```

Comment:
if both tables are keyed why not the result?

Reply:
I _think_ it's just being conservative: filed #7170 as follow-up for a possible improvement.

Reply:
I suspect tables might not be sorted, maybe they are rbind left and right